sync/downstream.py

```diff
@@ -107,7 +107,7 @@ def metadata_ready(self):
         latest_try_push = self.latest_valid_try_push
         if not latest_try_push:
             return False
-        if latest_try_push.status != "complete":
+        if latest_try_push.status != "complete" or latest_try_push.infra_fail:
```

Comment:
Maybe we also want to check for expired here?

Reply:
Done.